pkg/pool-hooks/contracts/StableSurgeHookExample.sol

```diff
@@ -0,0 +1,243 @@
+
+        return true;
+    }
+
+    /// @inheritdoc IHooks
+    function onComputeDynamicSwapFeePercentage(
+        PoolSwapParams calldata params,
+        address pool,
+        uint256 staticSwapFeePercentage
+    ) public view override onlyVault returns (bool, uint256) {
+        uint256 amp;
+        (amp, , ) = IStablePool(pool).getAmplificationParameter();
+
+        // In order to calculate `weightAfterSwap` we need balances after swap, so we must compute the swap amount.
+        uint256 invariant = StableMath.computeInvariant(amp, params.balancesScaled18);
+        uint256 amountCalculatedScaled18;
+
+        if (params.kind == SwapKind.EXACT_IN) {
+            amountCalculatedScaled18 = StableMath.computeOutGivenExactIn(
+                amp,
+                params.balancesScaled18,
+                params.indexIn,
+                params.indexOut,
+                params.amountGivenScaled18,
+                invariant
+            );
+
+            // Swap fee is always a percentage of the amountCalculated. On ExactIn, subtract it from the calculated
+            // amountOut. Round up to avoid losses during precision loss.
+            uint256 swapFeeAmountScaled18 = amountCalculatedScaled18.mulUp(staticSwapFeePercentage);
+            amountCalculatedScaled18 -= swapFeeAmountScaled18;
+        } else {
+            amountCalculatedScaled18 = StableMath.computeInGivenExactOut(
+                amp,
+                params.balancesScaled18,
+                params.indexIn,
+                params.indexOut,
+                params.amountGivenScaled18,
+                invariant
+            );
+
+            // To ensure symmetry with EXACT_IN, the swap fee used by ExactOut is
+            // `amountCalculated * fee% / (100% - fee%)`. Add it to the calculated amountIn. Round up to avoid losses
+            // during precision loss.
+            uint256 swapFeeAmountScaled18 = amountCalculatedScaled18.mulDivUp(
+                staticSwapFeePercentage,
+                staticSwapFeePercentage.complement()
+            );
+
+            amountCalculatedScaled18 += swapFeeAmountScaled18;
+        }
+
+        return (
+            true,
+            getSurgeFeePercentage(
+                params,
+                amountCalculatedScaled18,
+                _surgeThresholdPercentage[pool],
+                staticSwapFeePercentage
+            )
+        );
+    }
+
+    /**
+     * @notice Sets the hook threshold percentage.
+     * @dev This function must be permissioned. If the pool does not have a swap fee manager role set, the surge
+     * threshold will be effectively immutable, set to the default threshold for this hook contract.
+     */
+    function setSurgeThresholdPercentage(address pool, uint256 newSurgeThresholdPercentage) external {
+        if (_vault.getPoolRoleAccounts(pool).swapFeeManager != msg.sender) {
+            revert SenderNotAllowed();
+        }
```

Comment:
This should probably follow the same logic as the swap fee percentage: if there's no swap fee manager, default to governance.